### PR TITLE
core: always set output arg in unit_status_string()

### PR DESCRIPTION
**Pull request by keszybz**
As requested in https://github.com/systemd/systemd/pull/20058#pullrequestreview-696942153.

### CODE REVIEW

File: src/core/unit.c [hunk 1]
```diff
@@ -1340,27 +1340,34 @@ const char* unit_description(Unit *u) {
...
-                } else
-                        log_oom();
+        if (u->manager->status_unit_format == STATUS_UNIT_FORMAT_COMBINED &&
+            ret_combined_buffer) {
+
+                *ret_combined_buffer = strjoin(u->id, " - ", u->description);
+                if (*ret_combined_buffer)
+                        return *ret_combined_buffer;
+                log_oom(); /* Fall back to ->description */
         }
```
Comment (poettering) [inline src/core/unit.c:1372]: this is missing an else branch, no:

```c
else if (*ret_combined_buffer)
         *ret_combined_buffer = NULL;
```

**Comment by keszybz** [comment]
Duh, updated.

**Review by bluca — APPROVED** [review]
LGTM

**Comment by poettering** [comment]
lgtm

**Pull request merged**